rpcs3/util/shared_ptr.hpp

```diff
@@ -603,9 +605,10 @@ namespace stx
 
 		template <typename T1>
 		static constexpr bool is_stx_pointer = false
-			|| is_instance_of<std::remove_cv_t<T1>, shared_ptr>::value
-			|| is_instance_of<std::remove_cv_t<T1>, single_ptr>::value
-			|| is_instance_of<std::remove_cv_t<T1>, atomic_ptr>::value;
+			|| is_instance_of<std::remove_cvref_t<T1>, shared_ptr>::value
+			|| is_instance_of<std::remove_cvref_t<T1>, single_ptr>::value
+			|| is_instance_of<std::remove_cvref_t<T1>, atomic_ptr>::value;
+			|| std::is_same_v<std::remove_cvref_t<T1>, null_ptr_t>;
```

Comment:
We need to thank C++ gods that std::is_same accepts incomplete types for this ;)